### PR TITLE
Builds with Xcode 8 and adapts a bit to new devices

### PR DESCRIPTION
**Pull request by mgmart**
- DropboxSDK updated (the old one was still 32bit)
- Changed storage of AppKey because old method collided with cocoapods 
   - if necessary cocoapods could be added. At the moment there are none
- Removed GHUnit-Test framework
   - tests have been migrated to XCTest
- Nowadays user-defaults arrays are not mutable
  - made sure that mutable copy is instantiated where needed
- Capture badge bug which was fixed in master is also fixed here
- Added Launchscreen and removed dummy-pngs
- RegexKitLite was removed because of old and not maintained
  - it caused a lot of warnings, also
- Some warnings were eliminated - still many there
- Moved pngs out of root directory

### CODE REVIEW

**Comment by webframp** [comment]
This is great to see, thanks for contributing! I'll spend some time experimenting with it tonight

**Comment by webframp** [comment]
Build failed, but it's just due to code signing identities, I'll check that out too

**Comment by mgmart** [comment]
Repaired buildscript. Included unit test-run. 🤓

**Comment by webframp** [comment]
Very nice, ran into some issues due to updates on my dev machine last night but I'll pull these latest changes and build

**Comment by webframp** [comment]
Initial testing is looking great, only 74 warnings 😁 
I will spend a little time exploring on device to make sure, then I can try to get a beta out

**Comment by webframp** [comment]
Some warning preventing submission to test flight right now, I'll need to sort through those

**Comment by mgmart** [comment]
Xcode had recommended to add one missing graphic, maybe this helps with the other warnings to.
Compile time warnings are now down to 37 🤓

**Comment by d12frosted** [comment]
@mgmart it fails to build 😸 Fails for me locally with the same error as in travis.

**Comment by mgmart** [comment]
@d12frosted yeah, thought it was hickup :) Got it (I hope) 🤓

**Comment by webframp** [comment]
Did you find you needed to update AlamoFire to 4.0 ? I was getting swift compile errors using an older one, see pod install output here: https://gist.github.com/webframp/975e4e8741e045daeb963646fa1fc32d#file-pod_install-txt

**Comment by webframp** [comment]
Also seeing 'fatal error: cannot increment beyond endIndex' right now with captureComponentsMatchedBy regex func during initial sync operation, looking into it

**Comment by webframp** [comment]
At this point I'm able to build locally in Xcode 8.2 using pods Alamofire (4.0.1) and SwiftyDropbox (4.1.1). There's some bugs and warnings to still clean up, but it's definitely improving.

Since this targets the develop branch and builds also on Travis I would feel ok merging this and opening new PRs for fixing the additional bugs as we work towards a new release.

@mgmart What do you think?

File: Classes/Extensions/String+RegexSplitter.swift [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    do {
+      var result:[String] = []
+      let swiftString = String(self)
+      let rgx = try NSRegularExpression(pattern: regex, options: [])
+
+      let matches = rgx.matches(in: swiftString, options: [], range: NSRange(location: 0, length: swiftString.characters.count))
+      for match in matches {
+        for n in 0..<match.numberOfRanges {
+          let range = match.rangeAt(n)
+          let begin = swiftString.index(swiftString.startIndex, offsetBy: range.location)
```
Comment (webframp) [inline Classes/Extensions/String+RegexSplitter.swift]: This line consistently crashes for me with "fatal error: cannot increment beyond endIndex"

Comment (webframp) [inline Classes/Extensions/String+RegexSplitter.swift]: I think it's related to org file content, not directly the function logic, but is probably a case we need to need handle safely.

File: Classes/Sync/Dropbox/DropboxTransferManager.swift [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+
+  static let instance = DropboxTransferManager()
+
+  override init() {
+    let filePath = Bundle.main.path(forResource: "AppKey", ofType: "plist")
+    let plist = NSDictionary(contentsOfFile:filePath!)
+    let dkist = plist?["Dropbox API Key"] as? NSDictionary
+    let appKey = dkist?.object(forKey: "AppKey") as! String
+    DropboxClientsManager.setupWithAppKey(appKey)
```
Comment (d12frosted) [inline Classes/Sync/Dropbox/DropboxTransferManager.swift]: When I run this on my phone it end up with following crush.

```
fatal error: unexpectedly found nil while unwrapping an Optional value
2016-12-18 11:52:43.491941 MobileOrg[2940:909018] fatal error: unexpectedly found nil while unwrapping an Optional value
```

The last piece of code before some hell is following line in `OAuth.swift` in `SwiftyDropbox`.

```
self.dauthRedirectURL = URL(string: "db-\(appKey)://1/connect")!
```

**Comment by d12frosted** [comment]
Ok, it works for me 😸 

P. S. what about using [these loading images](https://github.com/d12frosted/mobileorg/commit/cfab4f5525c583a8eaa009f9024e952ce3922c56)?

**Comment by mgmart** [comment]
@d12frosted saw the loading images before in your fork. Thought about investigating how to use pdf for launch screens before adding it.

**Comment by mgmart** [comment]
@webframp I do not have adhoc, debug or release targets in the project, so can't reproduce the podfile version problem.

I think it's a good approach to merge this huge PR and then work on smaller chunks.

**Comment by webframp** [comment]
@d12frosted I think adding the loading images would be great, but lets do it in a dedicated PR

**Comment by webframp** [comment]
@mgmart ok, I'm ready to merge this then

**Comment by d12frosted** [comment]
> I think adding the loading images would be great, but lets do it in a dedicated PR

Agree. I'll fire another PR for this then.

**Pull request merged**